src/Pdf.php

```diff
@@ -125,6 +125,31 @@ public function saveImage($pathToImage)
     }
 
     /**
+     * Save the file as images to the given path.
+     *
+     * @param string $storeDirectory
+     * @param string $file_name     
+     *
+     * @return array $files the full path to the newly created images.
+     */
+    public function extractAllPagesAsImages($storeDirectory, $file_name)
+    {
+        $pages = $this->getNumberOfPages();
+        $files = [];
+        for($i =1; $i <= $pages; $i++)
+        {
+            $this->setPage($i);
+            $file_path = "{$storeDirectory}/page_{$i}_{$file_name}.{$this->outputFormat}";
+            $imagick = $this->getImageData($file_path);
```

Comment:
`$imagick` contains the image data. It's better to call this variable `$imageData` 😄